framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4520,4 +4518,13 @@ public boolean containsSameByClass(
         }
         return null;
     }
+
+    /**
+     * Returns a WholeProgramInference for use by this type factory.
```

Comment:
This method creates the `WholeProgramInference`. `getWholeProgramInference` returns it.  Also, I think this method should be `protected`.